src/formatter/renameField.ts

```diff
@@ -0,0 +1,28 @@
+import * as E from "fp-ts/Either";
+import * as O from "fp-ts/Option";
+import { modifyAt } from "fp-ts/Record";
+import { identity, pipe } from "fp-ts/function";
+
+const jsonParse: <T>(
+  text: string
+) => E.Either<Error, Record<string, T>> = E.tryCatchK(JSON.parse, E.toError);
+
+export const renameField = <T>(
+  input: string,
+  field: keyof Record<string, T>,
+  newField: T
+): string =>
+  pipe(
+    jsonParse<T>(input),
+    E.bimap(
+      err => `Cannot parse JSON: ${err.message}`,
+      parsedJson =>
+        pipe(
+          parsedJson,
+          modifyAt(field, () => newField),
+          O.getOrElseW(() => `Cannot find field: ${field}`),
+          JSON.stringify
+        )
+    ),
+    E.getOrElse(identity)
+  );
```

Comment:
In order to have` input `as generic with T type and be sure that `field` is related to `input ` (keyof T) I think we need to have a generic function like this . What do you think @CrisTofani ? Thanks!

```suggestion
export const renameField = <T extends Record<string, string | number>>(
  input: T,
  field: keyof T,
  newField: string | number
): Record<string, string | number> | string =>
  pipe(
    input,
    modifyAt(String(field), () => newField),
    O.getOrElseW(() => `Cannot find field: ${String(field)}`)
  );
```

Reply:
Made two changes:

- the function now returns an Either<Error, T> instead of Record<string, unknown> | string
- T extends Record<string, unknown> instead of Record<string, string | number>

What do you think @AleDore ?

```suggestion
import * as E from "fp-ts/Either";
import * as O from "fp-ts/Option";
import { modifyAt } from "fp-ts/Record";
import { pipe } from "fp-ts/function";


export const renameField = <T extends Record<string, unknown>>(
  input: T,
  field: keyof T,
  newField: unknown
): E.Either<Error, T> =>
  pipe(
    input,
    modifyAt(String(field), () => newField),
    O.match(
      () => E.left(new Error(`Cannot find field: ${String(field)}`)),
      modifiedObject => E.right(modifiedObject as T)
    )
  );
```

Reply:
Instead of using the `O.match` utility, what about 

```
    E.fromOption(() => new Error(`Field ${String(field)} not found`)),
    E.map(r => r as T)
```

I added the last instruction to bypass a compilation error, but I'm not totally sure about it, if there's a better option feel free to replace